From c842cb3306ab728c9c4b3aea553a10f18d383089 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 22 Jan 2024 17:18:23 +0200
Subject: [PATCH] [SW-170792] gaudi3: pif: drop pci vpd type 1 msgs

In previous patch the mctp messages coming from the host were blocked
in the MCTP block, and some interrupt had to be masked to allow that.
much cleaner solution is to block this kind of messages in the pci
controller.
this patch replaces the previous solution and clears FILTER_MASK_2_OFF
register of the DBI space, which according to designware datasheet means
"vendor MSG type 1 dropped silently"

Signed-off-by: Igal Zeltser <izeltser@habana.ai>
Change-Id: I56bb2b23a9d7686bef3169e75ab3491bc9fa4dcd
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index e4b8de9..1fdb5b4 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1023,18 +1023,9 @@
 	/* remove Vital Product Data (VPD) reporting capability */
 	regmap_rmwritel(D0_PCIE_DBI_SIG_REG(mmPCIE_DBI_PCI_MSIX_CAP_ID_NEXT_CTRL_REG), 0,
 			PCIE_DBI_PCI_MSIX_CAP_ID_NEXT_CTRL_REG_PCI_MSIX_CAP_NEXT_OFFSET_M);
-}
 
-#define MCTP_QUEUE_DROP_EVT_MASK	BIT(2)
-
-static void pif_arc_config_mctp_params(void)
-{
-	/* mask HBW MCTP Queue Drop event */
-	regmap_writel(MCTP_QUEUE_DROP_EVT_MASK, mmD0_PCIE_WRAP_MCTP_BASE +
-						mmPCIE_WRAP_MCTP_INTR_MASK);
-	/* disable turning mctp msg to inbound transaction */
-	regmap_writel(PCIE_WRAP_MCTP_MCTP_IN_ENABLE_V_M, mmD0_PCIE_WRAP_MCTP_BASE +
-							 mmPCIE_WRAP_MCTP_MCTP_IN_ENABLE);
+	/* drop Vendor Defined Messages (VDM) type 1 */
+	regmap_writel(0, D0_PCIE_DBI_SIG_REG(mmPCIE_DBI_FILTER_MASK_2_OFF));
 }
 
 __weak void preboot_update_bar_sizes(void) {}
@@ -1050,8 +1041,6 @@
 	pif_arc_pcie_device_id_vendor_id_set();
 
 	pif_arc_config_dbi_params();
-
-	pif_arc_config_mctp_params();
 
 	/* Set HBM BAR size */
 	preboot_update_bar_sizes();
